test/unit/org/apache/cassandra/cql3/ast/Mutation.java

```diff
@@ -766,18 +847,32 @@ public UpdateBuilder set(Symbol column, Expression value)
 
         public UpdateBuilder set(String column, int value)
         {
-            return set(new Symbol(column, Int32Type.instance), Bind.of(value));
+            Symbol symbol = find(column);
+            if (!symbol.type().equals(Int32Type.instance))
+                throw new AssertionError("Expected int type but given " + symbol.type().asCQL3Type());
+            return set(symbol, Bind.of(value));
+        }
+
+        public UpdateBuilder set(String column, Object value)
+        {
+            Symbol symbol = find(column);
+            return set(symbol, new Bind(value, symbol.type()));
         }
 
         public UpdateBuilder set(String column, Expression expression)
         {
-            Symbol symbol = new Symbol(metadata.getColumn(new ColumnIdentifier(column, true)));
-            return set(symbol, expression);
+            return set(find(column), expression);
+        }
+
+        public UpdateBuilder set(String column, Function<Symbol, Expression> fn)
```

Comment:
Also unused?

Reply:
unused in this patch, but is used in the batch patch